tests/library/trace-viewer.spec.ts

```diff
@@ -558,6 +559,17 @@ test('should handle src=blob', async ({ page, server, runAndTrace, browserName }
   expect(size).toBe(10);
 });
 
+test('should handle file URIs', async ({ page, runAndTrace, browserName }) => {
+  test.skip(browserName !== 'chromium');
+
+  const traceViewer = await runAndTrace(async () => {
+    await page.goto(pathToFileURL(path.join(__dirname, '..', 'assets', 'one-style.html')).href);
+  });
+
+  const frame = await traceViewer.snapshotFrame('goto');
+  await expect(frame.locator('body')).toHaveCSS('background-color', 'rgb(255, 192, 203)');
+});
```

Comment:
```suggestion
test('should handle file URIs', async ({ page, runAndTrace }) => {
  const file = test.info().outputPath('file.html');
  await fs.promises.writeFile(file, '<div>Hello world! :)</div>');
  const traceViewer = await runAndTrace(async () => {
    await page.goto(pathToFileURL(file).href);
  });
  const frame = await traceViewer.snapshotFrame('goto');
  await expect(frame.locator('body')).toHaveText('Hello world! :)');
  await expect(traceViewer.page.locator('.browser-frame-address-bar')).toHaveText(pathToFileURL(file).href);
});
```

Do you know why it wasn't working in the other browsers? I'd otherwise just change it to a text assertion, like this one.

Reply:
The thing this PR fixes is specifically related to loading a subresource, like a stylesheet, from a file URI. I think a text assertion would have already worked regardless, since the DOM snapshot just contains the text directly (as opposed to containing a `<link>` to the stylesheet, which the trace viewer then needs to load).

On Firefox it doesn't look like the trace file contains the stylesheet data. I didn't look closely into why, but it seems like a problem with _recording_ the trace, not viewing it. With WebKit, the test passes locally for me but not in CI, and I didn't debug why.

Reply:
For future reference:

- Firefox does not issue a response for `file://` requests.
- WebKit does not return headers for `file://` requests and reports it with `status() = 0`